ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -804,6 +853,23 @@ export const ApplicationResourceTree = (props: ApplicationResourceTreeProps) =>
         }
     }, [props.filters]);
 
+    const [defaultCompactView, setDefaultCompactView] = React.useState(false);
+    React.useEffect(() => {
+        const {podGroupCount, setShowCompactNodes, appContext} = props;
+        const podCount = nodes.filter(node => node.kind === 'Pod').length;
+
+        if (!defaultCompactView && podCount > podGroupCount) {
+            setShowCompactNodes(true);
+            setDefaultCompactView(true);
+
+            appContext.apis.notifications.show({
+                content: `Since the number of pods has surpassed the threshold pod count of ${podGroupCount}, you will now be switched to the group node view.
+                 If you prefer the tree view, you can simply click on the group node icon to deselect the current view.`,
```

Comment:
...simply click on the group node icon to....

The button has a hover help of Group Nodes, so what do you think about:

...simply click on the Group Nodes toolbar button to....     ?